### ISSUE TITLE
Cannot register new account - code sent to other device

### ISSUE THREAD

**Issue by sunlit777** [TelegramBeta/Support#5]
New freshly purchased SIM card, official IOS app - cannot register because “code sent to other device” and sms@telegram.org never replied in 2 months!
The issue is country specific - for USA, Russia and (I presume others as well) “send by sms” option is readily available. But for some other countries’s numbers like +62 for Indonesia they DON’t offer SMS and “send to other NON-EXISTENT device”. No way to get hold of support either

**Comment by difo-n8r** [comment]
I have the same problem, after looking more in the internet. It seems the issue is not new and always affecting certain phone number. See [tdesktop #26626](https://github.com/telegramdesktop/tdesktop/issues/26626#issue-1838250788). When tinkering with official Telegram apps for Android, I saw that if you input random phone number, the _"code sent to other device"_ screen too. Could it be the Telegram apps failed to recognize recently purchased phone number? Anyway I hoped it gets fixed because I don't want to dispose my recently purchased phone number just because it can't sign up to Telegram.

**Comment by sunlit777** [comment]
I’ve figured everything out:

The main issue is that Telegram presents registration prompts differently in different countries. In some places like USA and Russia, the send sms option is always there. 

In other less fortunate localities, e.g. Indonesia (400 million people, Telegram very popular) Telegram just doesn’t give a crap about their users, and if you are “lucky” to get a phone number that was previously registered for Telegram, and not properly unregistered (that party might even still be using it, and then you are soo out of luck, because you cannot even hope for timeout of inactive account to auto-delete the account) you get stuck because you ARE NEVER presented with an option to get code via SMS. 
This is not a rare and unusual scenario because in a country of 400 million people cellular operators are CONSTANTLY ROTATING old expired phone numbers back into operation. To put it simply, if a guy buys SIM card, registers it for Telegram and than trashes it, the operator will reuse that phone number again in 6 months or so, and the next customer is in for trouble!
 In this situation the only option is to send email to ***@***.*** ***@***.***> complaining of not being able to get the code but NOBODY ever replied to me from that e-mail IN 3 MONTHS!
You can also send message to ***@***.*** ***@***.***> - WITH SAME RESULT - NO REPLY IN MONTHS.
You can also contact their Twitter/X official account - ALSO NOBODY THERE to take care of you.
Telegram support is 100% NON-EXISTENT, even for PREMIUM users!
I was able to solve my issue by contacting a work e-mail of some telegram employee that I found in some forums, possibly here on Gibhub. He was able to help and he reset the registration status on my number and I was able to get enrolled to Telegram. To my question why Telegram is doing it, he replied with silence…

So there you go guys, there is no solution for you if you are stuck with “code sent to other device” issue, until Telegram changes their policy.
Best luck

**Comment by iamhanafi** [comment]
Hi @sunlit777 , I have the same issue. I didn't receive the code, and there is no response from the Telegram support team. Can you please send me the email of that person? Here is my email 13s4w7j19@mozmail.com

I would really appreciate it if you could help me.
Thank you

**Comment by sunlit777** [comment]
sorry but that is a private email of a Telegram staff member. I can’t share
it. My advise is to put out more publicity about this issue. make it loud
so it starts hurting their business, otherwise they will continue to ignore
users from certain countries! Make it racially motivated, because in US and
Europe there’s always an option to get sms code

**Comment by iamhanafi** [comment]
Yes, I know. I already created a post about it, and many people have posted the same issue. I have no influence, and they don't care. I hate this shit but I think I'm stuck here wkwkwk....

**Comment by DarkJoney** [comment]
I am so angry!!!!

This "security feature" creates a catastrophic scenario when accounts are actively compromised. My father's account was stolen through a phishing attack using Telegram's login widget. The attacker gained a session on their device. When we tried to log back in from the legitimate phone, Telegram said "We've sent code to your other device" and provided no SMS option whatsoever. The attacker's active session completely blocks any SMS delivery to the legitimate owner.

The problem is that your "protection against SIM-swap attacks" becomes a tool FOR attackers. Once they have a session, they effectively lock the legitimate owner out permanently. There is no mechanism to force-terminate suspicious sessions without account access, no way to request an SMS code when you control the phone number, and no way to prove ownership through any alternative method.

We have the phone number and can receive SMS. We have the account history. We have no access to the recovery email because the attacker changed it. There is zero recovery path. We contacted [sms@telegram.org](mailto:sms@telegram.org) days ago with no response. Meanwhile the account is actively sending spam to all contacts, business communications are lost, and personal data is being harvested.

What should exist is a forced session reset after several hours of failed login attempts from the legitimate phone number, an SMS override option when the account shows suspicious activity like logins from new countries or email changes, carrier-level verification for users who can prove phone number ownership, and actual support response within 48 hours for hijacked accounts.

Your feature protects against hypothetical SIM-swap attacks which are rare, while enabling real-time account hijacking that's happening right now to thousands of users based on reports here and across forums. This is not a bug, this is a fundamental design flaw that prioritizes theoretical security over actual user safety. Users who control the registered phone number should be able to regain access to their accounts, but attackers with one stolen session can permanently lock out legitimate owners. 
This is broken.
